helix-vcs/src/git.rs

```diff
@@ -112,6 +147,69 @@ impl DiffProvider for Git {
 
         Ok(Arc::new(ArcSwap::from_pointee(name.into_boxed_str())))
     }
+
+    fn get_changed_files(
+        &self,
+        cwd: &Path,
+    ) -> Result<Box<dyn Iterator<Item = Result<FileChange>>>> {
+        Self::status(&Self::open_repo(cwd, None)?.to_thread_local())
+    }
+}
+
+impl From<Git> for DiffProviderImpls {
+    fn from(value: Git) -> Self {
+        DiffProviderImpls::Git(value)
+    }
+}
+
+impl std::iter::Iterator for StatusIter {
+    type Item = Result<FileChange>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mapper = |item: std::result::Result<Item, _>| {
+            Result::<Option<FileChange>>::Ok(match item? {
+                Item::Modification {
+                    rela_path, status, ..
+                } => {
+                    let full_path = self.work_dir.join(rela_path.to_path()?);
+
+                    match status {
+                        EntryStatus::Conflict(_) => {
+                            // We treat conflict as change for now for simplicity
+                            Some(FileChange::Modified { path: full_path })
+                        }
+                        EntryStatus::Change(change) => match change {
+                            Change::Removed => Some(FileChange::Deleted { path: full_path }),
+                            Change::Modification { .. } => {
+                                Some(FileChange::Modified { path: full_path })
+                            }
+                            // No submodule support for now
+                            Change::Type | Change::SubmoduleModification(_) => None,
+                        },
+                        EntryStatus::NeedsUpdate(_) | EntryStatus::IntentToAdd => None,
+                    }
+                }
+                Item::DirectoryContents { entry, .. } => Some(FileChange::Untracked {
```

Comment:
`entry.status` is something you'd definitely want to check and only use `Untracked` entries, for robustness and clarity.

Reply:
Thanks! Totally missed this. Fixed now.